packages/celotool/src/e2e-tests/utils.ts

```diff
@@ -385,109 +125,155 @@ export async function migrateContracts(
     '-t',
     to.toString(),
   ]
-  await execCmdWithExitOnFailure('yarn', args)
+  await spawnCmdWithExitOnFailure('yarn', args)
 }
 
-export function getContractAddress(contractName: string) {
-  const filePath = `${MonorepoRoot}/packages/protocol/build/testing/contracts/${contractName}.json`
-  const contractData = JSON.parse(fs.readFileSync(filePath, 'utf8'))
-  return contractData.networks[NetworkId].address
-}
-
-export async function snapshotDatadir(instance: GethInstanceConfig) {
-  // Sometimes the socket is still present, preventing us from snapshotting.
-  await execCmd('rm', [`${getDatadir(instance)}/geth.ipc`], { silent: true })
-  await execCmdWithExitOnFailure('cp', ['-r', getDatadir(instance), getSnapshotdir(instance)])
-}
-
-export async function restoreDatadir(instance: GethInstanceConfig) {
-  const datadir = getDatadir(instance)
-  const snapshotdir = getSnapshotdir(instance)
-  console.info(`geth:${instance.name}: restore datadir: ${datadir}`)
-  await execCmdWithExitOnFailure('rm', ['-rf', datadir], { silent: true })
-  await execCmdWithExitOnFailure('cp', ['-r', snapshotdir, datadir], { silent: true })
-}
-
-function getInstanceDir(instance: GethInstanceConfig) {
-  return joinPath(TEST_DIR, instance.name)
-}
-
-function getDatadir(instance: GethInstanceConfig) {
-  return joinPath(getInstanceDir(instance), 'datadir')
-}
+export async function startBootnode(
+  bootnodeBinaryPath: string,
+  mnemonic: string,
+  gethConfig: GethRunConfig,
+  verbose: boolean
+) {
+  const bootnodePrivateKey = getPrivateKeysFor(AccountType.BOOTNODE, mnemonic, 1)[0]
+  const bootnodeLog = joinPath(gethConfig.runPath, 'bootnode.log')
+  const bootnodeArgs = [
+    '--verbosity=5',
```

Comment:
Verbosity 5 eats disk space at a furious pace. We shouldn't make this the default.
```suggestion
    '--verbosity=4',
```